runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -478,8 +415,84 @@ <T, W extends BoundedWindow> void writePCollectionViewData(
         throws IOException;
   }
 
-  String getStateFamily(NameContext nameContext) {
-    return nameContext.userName() == null ? null : stateNameMap.get(nameContext.userName());
+  /**
+   * Execution states in Streaming are shared between multiple map-task executors. Thus this class
+   * needs to be thread safe for multiple writers. A single stage could have have multiple executors
+   * running concurrently.
+   */
+  public static class StreamingModeExecutionState
+      extends DataflowOperationContext.DataflowExecutionState {
+
+    // AtomicLong is used because this value is written in two places:
+    // 1. The sampling thread calls takeSample to increment the time spent in this state
+    // 2. The reporting thread calls extractUpdate which reads the current sum *AND* sets it to 0.
+    private final AtomicLong totalMillisInState = new AtomicLong();
+
+    // The worker that created this state.  Used to report lulls back to the worker.
+    @SuppressWarnings("unused") // Affects a public api
```

Comment:
can we remove the member variable and put the suppresion on the public constructor parameter instead?

Reply:
done.